manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
+	distribution.Descriptor
+	// ForeignURL contains the foreign source URL of this layer.
+	ForeignURL string `json:"foreign_url,omitempty"`
```

Comment:
In general, this works because `foreign` and URL doesn't need to be combined. More precisely, the presence of URL shouldn't imply the foreign media type and vice versa.